api/api.go

```diff
@@ -76,6 +76,14 @@ func (srv *Server) initAPI(addr string) {
 		handleHTTPRequest(mux, "/miner/submitblock", srv.minerSubmitblockHandler)
 		handleHTTPRequest(mux, "/miner/headerforwork", srv.minerHeaderforworkHandler)
 		handleHTTPRequest(mux, "/miner/submitheader", srv.minerSubmitheaderHandler)
+		handleHTTPRequest(mux, "/miner/connecttopool", srv.minerConnectToPoolHandler)
+		handleHTTPRequest(mux, "/miner/poolheaderforwork", srv.minerPoolHeaderForWorkHandler)
+		handleHTTPRequest(mux, "/miner/submitpoolheader", srv.minerSubmitPoolHeaderHandler)
```

Comment:
Under the new paradigm, I think we'd see the following calls:

/miner GET -> return the miner status
/miner/start POST -> start the miner
/miner/stop POST -> stop the miner

/miner/block GET -> get a block for work
/miner/block POST -> submit a block that was worked

/miner/header GET -> get a header for work
/miner/header POST -> submit a header for work

/miner/pool  PUT -> connect to a pool (or maybe /miner/pool/connect PUT)
/miner/pool GET -> get a pool header for work
/miner/pool POST -> submit a solved pool header